File: 30 Days of Code/Day 14 Scope/difference.cpp
#include <bits/stdc++.h>
using namespace std;

class Difference
{
	public:
		int maximumDifference;

		Difference(vector<int> arr):
			elements(arr),
			maximumDifference(0)
		{}

		void computeDifference() {
			for (int i = 0; i < elements.size()-1; i++) {
				for (int j = i+1; j < elements.size(); j++) {
					int diff = abs(elements[i]-elements[j]);
					if (diff > maximumDifference) {
						maximumDifference = diff;
					}
				}
			}
		}

	private:
		vector<int> elements;
};

int main()
{
	int N;
	cin >> N;

	vector<int> a;

	for (int i = 0; i < N; i++) {
		int e;
		cin >> e;

		a.push_back(e);
	}

	Difference d(a);

	d.computeDifference();

	cout << d.maximumDifference;

	return 0;
}

File: 30 Days of Code/Day 14 Scope/difference.py
class Difference:
	def __init__(self, a):
		self.__elements = a[:]
        self.maximumDifference = 0

    def computeDifference(self):
        for i in range(len(self.__elements)-1):
            for j in range(i+1, len(self.__elements)):
                diff = abs(self.__elements[i] - self.__elements[j])
                self.maximumDifference = diff if diff > self.maximumDifference \
                                              else self.maximumDifference

_ = input()
a = [int(e) for e in input().split(' ')]

d = Difference(a)
d.computeDifference()

print(d.maximumDifference)
